airflow/providers/google/cloud/hooks/bigquery.py

```diff
@@ -349,56 +358,49 @@ def get_dataset_tables(self, dataset_id: str, project_id: Optional[str] = None,
         :param page_token: (Optional) page token, returned from a previous call,
             identifying the result set.
         :type page_token: str
-
-        :return: map containing the list of tables + metadata.
+        :param retry: How to retry the RPC.
+        :type retry: google.api_core.retry.Retry
+        :return: List of tables associated with the dataset.
         """
-        service = self.get_service()
-
-        optional_params = {}  # type: Dict[str, Union[str, int]]
-        if max_results:
-            optional_params['maxResults'] = max_results
-        if page_token:
-            optional_params['pageToken'] = page_token
-
-        dataset_project_id = project_id or self.project_id
+        project_id = project_id or self.project_id
 
-        return (service.tables().list(  # pylint: disable=no-member
-            projectId=dataset_project_id,
-            datasetId=dataset_id,
-            **optional_params).execute(num_retries=self.num_retries))
+        tables = Client(client_info=self.client_info).list_tables(
```

Comment:
client_info object does not contain all required data. BigQuery Client uses two connection protocols:
* GRPC - It works fine.
* HTTP for mulipart upload - It's doesn't contain user-agent.
Can you add user-agent to client_info object?

Reply:
I checked again. This information is available but is not at the beginning of the user agent.
```
gl-python/3.6.10 grpc/1.28.1 gax/1.17.0 gccl/airflow_v2.0.0.dev0
```

Reply:
So, are we ok with this as it is or should we explicitly pass user-agent?

Reply:
It's still correct. We don't need to change it.